datafusion/core/src/physical_plan/union.rs

```diff
@@ -240,14 +239,20 @@ impl ExecutionPlan for UnionExec {
         // which is the "meet" of all input orderings. In this example, this
         // function will return vec![false, true, true], indicating that we
         // preserve the orderings for the 2nd and the 3rd children.
-        self.inputs()
-            .iter()
-            .map(|child| {
-                ordering_satisfy(self.output_ordering(), child.output_ordering(), || {
-                    child.equivalence_properties()
+        if let Some(output_ordering) = self.output_ordering() {
+            self.inputs()
+                .iter()
+                .map(|child| {
+                    if let Some(child_ordering) = child.output_ordering() {
```

Comment:
I don't understand why we no longer need to check that the output order of the child is compatible with the output order of the input. Shouldn't this be calling `get_meet_of_orderings` to check rather than checking the length?

Reply:
The call `self.output_ordering()` in turn calls `get_meet_of_orderings` and gets the common longest prefix. We then check whether any of the inputs are equal to that prefix. We are checking for strict equality instead of an equivalence-aware check following @mingmwang's suggestion.